### PR TITLE
Add new conditions to Visibility Conditions

### PR DESCRIPTION
**Pull request by arinaturcu**
#743, #750, #752

### CODE REVIEW

**Comment by HardeepAsrani** [comment]
@arinaturcu Yesterday I left a couple of reviews above, I suppose I didn't mention your name so you didn't get the notification. The one about query string being a textarea needs to be done.

Also you can make Post Type as a Pro option.

I'm confused if Post Category should display if the page has all the selected categories or any of the selected. Any idea on how it's being done in Neve? We can make it consistent with Neve.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]

**Comment by arinaturcu** [comment]
@HardeepAsrani Thank you for the review! :rocket: 

> I'm confused if Post Category should display if the page has all the selected categories or any of the selected. Any idea on how it's being done in Neve? We can make it consistent with Neve.

I looked a bit into Neve and it seems that just one category can be selected in a condition, but maybe I am missing something.
If you think that would be better if we displayed the block if it has any of the selected categories, let me know.

I managed all the other things you mentioned, hope I didn't miss anything. :eyes:

File: src/blocks/plugins/conditions/edit.js [hunk 1]
```diff
@@ -250,6 +250,10 @@ const Edit = ({
 			attrs.meta_compare = 'is_true';
 		}
 
+		if ( 'queryString' === value ) {
```
Comment (HardeepAsrani) [inline src/blocks/plugins/conditions/edit.js:253]: Just a note that I've added the default value here. It was not working without it unless the user first changes the match value from one to another.

**Comment by HardeepAsrani** [comment]
When you get time, can you @irinelenache QA this too? Thanks!

**Comment by irinelenache** [comment]
@arinaturcu @HardeepAsrani Tested the features, here are my findings:

✅ The **Query string** conditions work as expected
✅ The **Post category** conditions work as expected

- [ ] For the **Post type** conditions, when i select more post types like [here](https://vertis.d.pr/3psDoo), the block is visible when only one of the selected post types is present, not both. (For the category conditions, all categories selected should be present in order for the block to be visible)

I couldn't test the visibility with blocks added as widgets. When i try to add any widget, i get an error in console https://vertis.d.pr/YWWDF0

Can be checked here:
```
 Admin area URL: https://likeswing.tastewp.com/wp-admin 
 Username: irinel 
 Password: aVUHGPPi3vQ 
```

Let me know if i missed anything, thank you 🚀

**Comment by HardeepAsrani** [comment]
> I couldn't test the visibility with blocks added as widgets. When i try to add any widget, i get an error in console https://vertis.d.pr/YWWDF0

It should be fixed now. It was fixed in v2 branch so I've rebased this branch.

**Comment by arinaturcu** [comment]
@irinelenache I think that a post can have a single type so it would not make much sense to check if a post has all the selected types.

@HardeepAsrani Please, let me know if I am wrong.
If I am right, then I will change the help text to something more suggestive.

**Comment by HardeepAsrani** [comment]
@arinaturcu I agree with that. If people want, they can add multiple conditions to force that. Everything else works as expected @irinelenache?

**Pull request merged**